dev/Dockerfile.test

```diff
@@ -6,22 +6,34 @@ ENV DEBIAN_FRONTEND=noninteractive \
 
 WORKDIR /app
 
-ADD . /app
-
 RUN apt-get update && \
-    apt-get install -y --no-install-recommends build-essential openjdk-11-jre-headless \
+    apt-get install -y --no-install-recommends build-essential openjdk-17-jre-headless \
     # cmake and protobuf-compiler required for onnx install
     cmake protobuf-compiler && \
     # Install miniforge
     wget -q https://github.com/conda-forge/miniforge/releases/latest/download/$MINIFORG_FILENAME && \
     bash $MINIFORG_FILENAME -b -p /opt/conda && \
-    rm $MINIFORG_FILENAME && \
+    rm $MINIFORG_FILENAME
+
+ADD ./requirements /app/requirements
```

Comment:
Build performance improvements:
- Move `ADD` commands later in the build.
- For `pip install`, add _only_ the `requirements/` directory.